### PR TITLE
fix c++ exception

### LINKED ISSUE #3927: C++ exception at doors.lua
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [ ] This issue is reproducible without changes to the C++ code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Start server
2. See error in console
3. Doors not working, but server working fine

### Error
```
\data\scripts\actions\others\doors.lua
C++ exception
> doors.lua [error]
```

### Reproduction environment
I can't test on clean tfs right now, can someone else confirm if this is happening to everyone? windows 10, compiled on release/64x

### PR DESCRIPTION
**Pull request by nekiro**
Fix #3927

### CODE REVIEW

**Review by EPuncker — APPROVED** [review]

**Review by DSpeichert — DISMISSED** [review]

**Review by EPuncker — DISMISSED** [review]

File: src/luascript.h [hunk 1]
```diff
@@ -299,24 +299,24 @@ class LuaScriptInterface
...
 		template<typename T>
-		static typename std::enable_if<(std::is_integral<T>::value && std::is_signed<T>::value) || std::is_floating_point<T>::value, T>::type
+		static typename std::enable_if<(std::is_integral<T>::value && (std::is_signed<T>::value) || std::is_floating_point<T>::value), T>::type
 			getNumber(lua_State* L, int32_t arg)
 		{
-			LUA_NUMBER luaNum = lua_tonumber(L, arg);
-			if (luaNum > std::numeric_limits<T>::max()) {
-				reportErrorFunc(L, fmt::format("Passed argument '{:s}' has invalid value: {:d}", arg, luaNum));
+			double num = lua_tonumber(L, arg);
+			if (num > static_cast<double>(std::numeric_limits<T>::max())) {
```
Comment (DSpeichert) [inline src/luascript.h:315]: Shouldn't you be checking for min value too?
```suggestion
			if (num > static_cast<double>(std::numeric_limits<T>::max()) || num < static_cast<double>(std::numeric_limits<T>::min())) {
```
Or something like that?

Comment (nekiro) [reply src/luascript.h:315]: I was, but it broke for some reason xD

Comment (DSpeichert) [reply src/luascript.h:315]: What broke?

**Review by dex-89 — DISMISSED** [review]

**Review by DSpeichert — APPROVED** [review]

File: src/luascript.h (hunk 1 above)
Comment (DSpeichert) [reply src/luascript.h]:
```
Lua Script Error: [Spell Interface]
data/spells/scripts/support/haste.lua
LuaScriptInterface::getNumber(). Passed argument '5' has invalid value: -24
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/haste.lua:7: in main chunk

Lua Script Error: [Spell Interface]
data/spells/scripts/support/haste.lua
LuaScriptInterface::getNumber(). Passed argument '3' has invalid value: -24
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/haste.lua:7: in main chunk

Lua Script Error: [Test Interface]
data/spells/scripts/support/sharpshooter.lua
LuaScriptInterface::getNumber(). Passed argument '4' has invalid value: -0.7
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/sharpshooter.lua:14: in main chunk
```

**Pull request merged**